### PR TITLE
Add Result.Create(...) methods (#85)

### PR DESCRIPTION
**Pull request by ahryshchanka**
Pull Request for resolve issue #85

### CODE REVIEW

File: CSharpFunctionalExtensions.Tests/ResultTests/ResultTests.cs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            Result result = Result.Create(false, "simple result error");
+
+            result.IsFailure.Should().BeTrue();
+            result.Error.Should().Be("simple result error");
+        }
+        
+        [Fact]
+        public void Create_predicate_is_true_Success_result_expected()
+        {
+            var predicate = new Mock<Func<bool>>();
```
[the remark this answers is not in the document]
Comment (ahryshchanka) [reply CSharpFunctionalExtensions.Tests/ResultTests/ResultTests.cs]: Ok

[the remark this answers is not in the document]
Comment (vkhorikov) [reply CSharpFunctionalExtensions.Tests/ResultTests/ResultTests.cs]: @ahryshchanka please do. No need for mocks in there either.

[the remark this answers is not in the document]
Comment (ahryshchanka) [reply CSharpFunctionalExtensions.Tests/ResultTests/ResultTests.cs]: Fixed in PR https://github.com/vkhorikov/CSharpFunctionalExtensions/pull/88

[the remark this answers is not in the document]
Comment (vkhorikov) [reply CSharpFunctionalExtensions.Tests/ResultTests/ResultTests.cs]: Thanks. I'll push a new version to nuget sometime early next week.

**Pull request merged**